primitives/src/merkle_tree/field_based_mht/path.rs

```diff
@@ -364,17 +358,22 @@ impl<T: FieldBasedMerkleTreeParameters> From<FieldBasedBinaryMHTPath<T>> for Fie
     }
 }
 
-impl<T: FieldBasedMerkleTreeParameters> From<FieldBasedMHTPath<T>> for FieldBasedBinaryMHTPath<T> {
-    fn from(other: FieldBasedMHTPath<T>) -> Self {
+impl<T: FieldBasedMerkleTreeParameters> TryFrom<FieldBasedMHTPath<T>> for FieldBasedBinaryMHTPath<T> {
+    type Error = Error;
+
+    fn try_from(other: FieldBasedMHTPath<T>) -> Result<Self, Self::Error> {
         let mut converted = Vec::with_capacity(other.path.len());
         for (nodes, position) in other.path {
-            // TODO: possible crash
-            assert!(nodes.len() == 1);
-            assert!(position == 0 || position == 1);
+            if nodes.len() != 1 {
+                Err(format!("FieldBasedMHTPath to FieldBasedBinaryMHTPath convert failed"))?
+            }
+            if position != 0 && position != 1 {
+                Err(format!("FieldBasedMHTPath to FieldBasedBinaryMHTPath convert failed"))?
```

Comment:
"Position must be only 0 or 1 for each element in the path to be able to perform conversion to a binary path"